packages/fxa-settings/src/pages/Signin/SigninRecoveryCode/container.test.tsx

```diff
@@ -0,0 +1,195 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as ReachRouterModule from '@reach/router';
+import * as ReactUtils from 'fxa-react/lib/utils';
+import * as CacheModule from '../../../lib/cache';
+import * as SigninRecoveryCodeModule from './index';
+
+import { MockedProvider, MockedResponse } from '@apollo/client/testing';
+import { loadErrorMessages, loadDevMessages } from '@apollo/client/dev';
+import { LocationProvider } from '@reach/router';
+import { renderWithLocalizationProvider } from 'fxa-react/lib/test-utils/localizationProvider';
+import SigninRecoveryCodeContainer from './container';
+import { createMockWebIntegration } from '../../../lib/integrations/mocks';
+import { MozServices } from '../../../lib/types';
+import { Integration } from '../../../models';
+import {
+  MOCK_STORED_ACCOUNT,
+  MOCK_RECOVERY_CODE,
+  mockLoadingSpinnerModule,
+} from '../../mocks';
+import { SigninRecoveryCodeProps } from './interfaces';
+import { mockGqlError, mockSigninLocationState } from '../mocks';
+import { mockConsumeRecoveryCodeUseMutation } from './mocks';
+import { waitFor } from '@testing-library/react';
+import { AuthUiErrors } from '../../../lib/auth-errors/auth-errors';
+
+let integration: Integration;
+function mockWebIntegration() {
+  integration = createMockWebIntegration() as Integration;
+}
+
+jest.mock('../../../lib/glean', () => ({
+  __esModule: true,
+  default: {
+    loginBackupCode: {
+      view: jest.fn(),
+      submit: jest.fn(),
+      success: jest.fn(),
+    },
+    isDone: jest.fn(),
+  },
+}));
+
+jest.mock('../../../models', () => {
+  return {
+    ...jest.requireActual('../../../models'),
+    useAuthClient: jest.fn(),
+  };
+});
+
+let currentSigninRecoveryCodeProps: SigninRecoveryCodeProps | undefined;
+function mockSigninRecoveryCodeModule() {
+  currentSigninRecoveryCodeProps = undefined;
+  jest
+    .spyOn(SigninRecoveryCodeModule, 'default')
+    .mockImplementation((props) => {
+      currentSigninRecoveryCodeProps = props;
+      return <div>signin recovery code mock</div>;
+    });
+}
+
+export function mockCache(opts: any = {}, isEmpty = false) {
```

Comment:
Does this need to be exported?

Reply:
Not needed, I'll remove. I initially had these in a separate mock file then opted to move them back in and missed the update.